src/datreant/core/collections.py

```diff
@@ -148,26 +148,29 @@ def __getitem__(self, index):
         select out members.
 
         """
+        memberlist = self._list()
         # we can take lists of indices, names, or uuids; these return a
         # View; repeats already not respected since View functions as a
         # set
-        if ((isinstance(index, list) or hasattr(index, 'dtype')) and
-                all([isinstance(item, bool) for item in index])):
+        if ((isinstance(index, list) and
+             all(isinstance(item, bool) for item in index)
+             or
+             hasattr(index, 'dtype') and index.dtype == 'bool')):
```

Comment:
the `and` will be evaluated before the `or` I would still prefer to use brackets though to make it explicit. https://docs.python.org/2/reference/expressions.html#operator-precedence